addons/hr_holidays/tests/test_uninstall.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.base.models.ir_model import MODULE_UNINSTALL_FLAG
+from odoo.tests import tagged, TransactionCase
+
+
+@tagged('-at_install', 'post_install')
+class TestHrLeaveUninstall(TransactionCase):
+    def test_unlink_model(self):
+        model = self.env['ir.model'].search([('model', '=', 'hr.leave')])
+        activity_type = self.env['mail.activity'].search([
+            ('res_model', '=', 'hr.leave')
+        ]).activity_type_id
+
+        self.assertTrue(activity_type)
+        self.assertIn('hr.leave', activity_type.mapped('res_model'))
+
+        model.sudo().with_context(**{MODULE_UNINSTALL_FLAG: True}).unlink()
+        self.assertEqual(model.exists(), self.env['ir.model'])
```

Comment:
This assertion is mostly useless: if you can `unlink()` a record, it's gone.
Please add assertions about `mail.activity`, `mail.activity.type`, and so on:
```suggestion
        self.assertFalse(model.exists())
        self.assertFalse(self.env['mail.activity'].search([('res_model', '=', 'hr.leave')]))
        self.assertFalse(self.env['mail.activity.type'].search([('res_model', '=', 'hr.leave')]))
        self.assertFalse(self.env['mail.followers'].search([('res_model', '=', 'hr.leave')]))
        self.assertFalse(self.env['mail.message'].search([('model', '=', 'hr.leave')]))
```

Reply:
I guess it makes the test more explicit and understandable (which is a good thing of course), but strictly speaking it isn't necessary because of the restricting ondelete trigger?

Reply:
I'm with Raphael here. Tests should be explicit to the thing they should check. Adding boilerplate to them only complicates the trace of the problems when they fail. Let each part to be self-tested and trust on the existing framework parts. There will be another test for what unlink means in the framework.